coc/client.py

```diff
@@ -278,14 +278,19 @@ def login_with_keys(self, *keys: str) -> None:
 
         LOG.debug("HTTP connection created. Client is ready for use.")
 
-    def close(self) -> None:
+    def close_loops(self) -> None:
         """Closes the HTTP connection
         """
         LOG.info("Clash of Clans client logging out...")
         self.dispatch("on_client_close")
         self.loop.run_until_complete(self.http.close())
         self.loop.close()
 
+    async def close_client(self) -> None:
+        """Closes the HTTP connection from within a loop function such as
+        async def main()"""
+        await self.http.close()
```

Comment:
```suggestion
        LOG.info("Clash of Clans client logging out...")
        self.dispatch("on_client_close")
        await self.http.close()
```
`on_client_close` event should dispatch here instead of in `close_loops`